packages/nextra-theme-docs/src/components/search.tsx

```diff
@@ -241,85 +117,77 @@ export function Search({
-            transition: 'max-height .2s ease' // don't work with tailwindcss
-          }}
-        >
-          {error ? (
-            <span className="_flex _select-none _justify-center _gap-2 _p-8 _text-center _text-sm _text-red-500">
-              <InformationCircleIcon className="_size-5" />
-              {renderString(themeConfig.search.error)}
-            </span>
-          ) : loading ? (
-            <span className="_flex _select-none _justify-center _gap-2 _p-8 _text-center _text-sm _text-gray-400">
-              <SpinnerIcon className="_size-5 _animate-spin" />
-              {renderComponent(themeConfig.search.loading)}
-            </span>
-          ) : results.length > 0 ? (
-            results.map(({ route, prefix, children, id }, i) => (
-              <Fragment key={id}>
-                {prefix}
-                <li
-                  className={cn(
-                    '_mx-2.5 _break-words _rounded-md',
-                    'contrast-more:_border',
-                    i === active
-                      ? '_bg-primary-500/10 _text-primary-600 contrast-more:_border-primary-500'
-                      : '_text-gray-800 contrast-more:_border-transparent dark:_text-gray-300'
-                  )}
-                >
-                  <NextLink
-                    className="_block _scroll-m-12 _px-2.5 _py-2"
-                    href={route}
-                    data-index={i}
-                    onFocus={handleActive}
-                    onMouseMove={handleActive}
-                    onClick={finishSearch}
-                    onKeyDown={handleKeyDown}
-                  >
-                    {children}
-                  </NextLink>
-                </li>
-              </Fragment>
-            ))
-          ) : (
-            renderComponent(themeConfig.search.emptyResult)
-          )}
-        </ul>
-      </Transition>
-    </div>
+        {error ? (
+          <span className="_flex _select-none _justify-center _gap-2 _p-8 _text-center _text-sm _text-red-500">
+            <InformationCircleIcon className="_size-5" />
+            {renderString(themeConfig.search.error)}
+          </span>
+        ) : loading ? (
+          <span className="_flex _select-none _justify-center _gap-2 _p-8 _text-center _text-sm _text-gray-400">
+            <SpinnerIcon className="_size-5 _animate-spin" />
+            {renderComponent(themeConfig.search.loading)}
+          </span>
+        ) : results.length ? (
+          results.map(({ route, prefix, children, id }) => (
+            <Fragment key={id}>
+              {prefix}
+              <ComboboxOption
+                as={NextLink}
+                value={route}
+                href={route}
+                className={cn(
+                  '_mx-2.5 _break-words _rounded-md',
+                  'contrast-more:_border',
+                  'data-[focus]:_ring data-[focus]:_bg-primary-500/10 data-[focus]:_text-primary-600 data-[focus]:contrast-more:_border-primary-500',
```

Comment:
```suggestion
                  'data-[focus]:_ring data-[focus]:_bg-primary-500/10 data-[focus]:!_text-primary-600 data-[focus]:contrast-more:_border-primary-500',
```

to change text primary color in dark mode too